src/compressed_tensors/quantization/lifecycle/helpers.py

```diff
@@ -29,17 +29,42 @@
 ]
 
 
-def update_layer_weight_quant_params(layer: Module):
-    weight = getattr(layer, "weight", None)
+def update_layer_weight_quant_params(
+    layer: Module,
+    weight: Optional[torch.Tensor] = None,
+    g_idx: Optional[torch.Tensor] = None,
+    reset_obs: bool = False,
+):
+    """
+    Update quantization parameters on layer
+
+    :param layer: input layer
+    :param weight: weight to update quant params with, defaults to layer weight
+    :param g_idx: optional mapping from column index to group index
+    :param reset_obs: reset the observer before calculating quant params,
+        defaults to False
+    """
+    attached_weight = getattr(layer, "weight", None)
+
+    if weight is None:
+        weight = attached_weight
     scale = getattr(layer, "weight_scale", None)
     zero_point = getattr(layer, "weight_zero_point", None)
+    if g_idx is None:
+        g_idx = getattr(layer, "weight_g_idx", None)
```

Comment:
Could you clarify the use case for casing on if weight and g_idx are None? In what case are we passing these values into this function vs grabbing them from the module itself?

Reply:
This function is only used by llm-compressor in the gptq pathway. After https://github.com/vllm-project/llm-compressor/pull/94 lands, the only use case will be passing in `weight` and `g_idx` directly.

However, I still want to leave the previous use case 1. for backwards compatibility and 2. so that llm-compressor still works after this PR lands

Reply:
got it!